src/lib/comms/sol-lwm2m.c

```diff
@@ -3013,7 +3003,15 @@ handle_resource(void *data, struct sol_coap_server *server,
         &content_format);
 
     if (r < 0)
-        content_format = SOL_LWM2M_CONTENT_TYPE_TEXT;
+        payload.type = SOL_LWM2M_CONTENT_TYPE_TEXT;
+    else
+        payload.type = content_format;
+
+    if (payload.type == SOL_LWM2M_CONTENT_TYPE_JSON) {
+        SOL_WRN("JSON content format is not supported");
+        header_code = SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
```

Comment:
According to Pg. 74 of the spec you should actually throw 4.15 Unsupported content format (`SOL_COAP_RESPONSE_CODE_UNSUPPORTED_CONTENT_FORMAT`).